lib/msal-browser/src/network/FetchClient.ts

```diff
@@ -37,11 +38,21 @@ export class FetchClient implements INetworkModule {
      */
     async sendPostRequestAsync<T>(url: string, options?: NetworkRequestOptions): Promise<NetworkResponse<T>> {
         const reqBody = (options && options.body) || "";
-        const response = await fetch(url, {
-            method: HTTP_REQUEST_TYPE.POST,
-            headers: this.getFetchHeaders(options),
-            body: reqBody
-        });
+
+        let response;
+        try {
+            response = await fetch(url, {
+                method: HTTP_REQUEST_TYPE.POST,
+                headers: this.getFetchHeaders(options),
+                body: reqBody
+            });
+        } catch (e) {
+            if (window.navigator.onLine) {
+                throw BrowserAuthError.createNetworkRequestFailedError(e, url);
+            } else {
+                throw BrowserAuthError.createNoNetworkConnectivityError();
+            }
+        }
         return {
             headers: this.getHeaderDict(response.headers),
             body: await response.json() as T,
```

Comment:
I'm curious if we should move this part into the try/catch, so that we can gracefully capture json parsing errors?

Reply:
Maybe a separate try catch so we can distinguish between parsing errors vs. network/CORS errors?

Reply:
Yeah that sounds good.